### PR TITLE
Preserve notation in AST

### PR DESCRIPTION
**Pull request by xtuc**
Closes https://github.com/xtuc/webassemblyjs/issues/209

I added a `raw` field to the `FloatLiteral`, `NumberLiteral` and `LongNumberLiteral` AST nodes and used it in the printer.

### CODE REVIEW

File: packages/wast-parser/test/fixtures/nan/canonical/expected.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
             {
               "type": "FloatLiteral",
               "value": 4194304,
```
[the remark this answers is not in the document]
Comment (maurobringolf) [reply packages/wast-parser/test/fixtures/nan/canonical/expected.json]: I introduced this in #85. The value is the payload carried by the `NaN`. This value should be the canonical one.

[the remark this answers is not in the document]
Comment (xtuc) [reply packages/wast-parser/test/fixtures/nan/canonical/expected.json]: ah yes sure, that make sense, thanks for the clarification.

**Comment by maurobringolf** [comment]
I just realized that numeric separators _ are lost here because the tokenizer completely ignores them. Not a big issue, but just saying in case it comes up somewhere later.

**Comment by xtuc** [comment]
That's a good point, could you please create an issue about that?

**Comment by maurobringolf** [comment]
Sure. It should be easy to fix by moving the deletion into the parse step instead of the tokenizer.

**Pull request merged**